homeassistant/helpers/service.py

```diff
@@ -112,3 +116,46 @@ def extract_entity_ids(hass, service_call, expand_group=True):
             return [service_ent_id]
 
         return service_ent_id
+
+
+@bind_hass
+def get_description(hass, domain, service):
+    """Return the description (i.e. user documentation) for a service call."""
+
+    if domain == ha.DOMAIN:
+        import homeassistant.components as components
+        file = components.__file__
+    else:
+        file = get_component(domain).__file__
+
+    descriptions = DESCRIPTION_CACHE.get(file)
+    if descriptions is None:
+        descriptions = load_yaml_config_file(
```

Comment:
This call will do I/O. Since `get_description` will be called from `async_get_all_descriptions`, the function that does I/O needs to be scheduled on the executor.

Reply:
Oh. This was moved from the LIFX `async_setup_platform` but I guess that was also wrong.

I am a bit confused by the main/executor transitions available. What is the proper way to schedule the function in this situation?

Reply:
I think the function that schedules the job for the executor, via `hass.async_add_job`, needs to be a coroutine to be able to yield and wait for the result, ie to have the descriptions ready. Not 💯 though. I'd wait for a core review before changing anything.

Reply:
You are correct Martin